### PR TITLE
Provide a name for the default demo datasets

### LINKED ISSUE #1124: Provide a name for the default demo datasets
### Problem Description
The default demo datasets are created on the fly and aren't named. All the other demo datasets have a name.

### Expected behavior
To make things clear and consistent, let's name the default demo datasets.

- For `load_tabular_demo`, let's name the default dataset `'demo_single_table'`
- For `load_demo`, let's name the default dataset `'demo_multi_table'`

If a user passes in these names, we should load the default datasets.

```python
from sdv.demo import load_tabular_demo, load_demo

single_table = load_tabular_demo(dataset_name='demo_single_table')
multi_table = load_demo(dataset_name='demo_multi_table')
```

### Additional context

For now, it is ok to keep the current functionality too. Ie. if a user does not provide a dataset name, then also return the demo dataset.

```python
from sdv.demo import load_tabular_demo, load_demo

single_table = load_tabular_demo()
multi_table = load_demo()
```

### PR DESCRIPTION
**Pull request by amontanez24**
resolves #1124

### CODE REVIEW

File: sdv/demo.py [hunk 1]
```diff
@@ -307,10 +308,11 @@ def load_demo(dataset_name=None, data_path=DATA_PATH, metadata=False):
             If ``metadata`` is ``False`` return a ``dict`` with the tables data.
             If ``metadata`` is ``True`` return a ``tuple`` with Metadata and tables data.
     """
-    if dataset_name:
-        meta, tables = _load_demo_dataset(dataset_name, data_path)
-    else:
+    dataset_name = dataset_name or 'demo_multi_table'
```
Comment (fealho) [inline sdv/demo.py:311]: I don't think you need this line.

File: sdv/demo.py [hunk 2]
```diff
@@ -375,7 +379,8 @@ def load_tabular_demo(dataset_name=None, table_name=None, data_path=DATA_PATH, m
             If ``metadata`` is ``False`` return a ``pandas.DataFrame`` with the tables data.
             If ``metadata`` is ``True`` return a ``tuple`` with a Table and the data.
     """
-    if dataset_name:
+    dataset_name = dataset_name or 'demo_single_table'
```
Comment (fealho) [inline sdv/demo.py:382]: Same

**Review by fealho — APPROVED** [review]
Looks good, just one comment.

**Review by pvk-developer — APPROVED** [review]
I'm on the same page as @fealho , otherwise lgtm!

**Pull request merged**